Assets/Tests/Weaver/Assembler.cs

```diff
@@ -179,6 +180,7 @@ public static AssemblyDefinition Build(IWeaverLogger logger)
             assemblyBuilder.buildFinished += delegate (string assemblyPath, CompilerMessage[] compilerMessages)
             {
                 CompilerMessages.AddRange(compilerMessages);
+#if !UNITY_2020_2_OR_NEWER
```

Comment:
can you move the `#if`  inside the foreach

Reply:
might also be worth removing `!` and swap the logic round since it is just an `#if/else`